ui/app/components/secret-engine/configure-azure.hbs

```diff
@@ -0,0 +1,103 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: BUSL-1.1
+~}}
+
+<form {{on "submit" (perform this.submitForm)}} aria-label="configure azure credentials" data-test-configure-form>
+  <NamespaceReminder @mode="save" @noun="configuration" />
+  <MessageError @errorMessage={{this.errorMessage}} />
+  <div class="box is-fullwidth is-sideless">
+    {{! accessType can be "azure" or "wif" - since WIF is an enterprise only feature we default to "azure" for community users and only display those related form fields. }}
+    {{#if this.version.isEnterprise}}
+      <fieldset class="field form-fieldset" id="protection" data-test-access-type-section>
+        <legend class="is-label">Access Type</legend>
+        <p class="sub-text" data-test-access-type-subtext>
+          {{#if this.disableAccessType}}
+            You cannot edit Access Type if you have already saved access credentials.
+          {{else}}
+            Choose the way to configure access to Azure. Access can be configured either using an Azure account or with the
+            Plugin Workload Identity Federation (WIF).
+          {{/if}}
+        </p>
+        <div>
+          <RadioButton
+            id="access-type-azure"
+            name="azure"
+            class="radio"
+            data-test-access-type="azure"
+            @value="azure"
+            @groupValue={{this.accessType}}
+            @onChange={{fn this.onChangeAccessType "azure"}}
+            @disabled={{this.disableAccessType}}
+          />
+          <label for="access-type-azure">Azure account credentials</label>
+          <RadioButton
+            id="access-type-wif"
+            name="wif"
+            class="radio has-left-margin-m"
+            data-test-access-type="wif"
+            @value="wif"
+            @groupValue={{this.accessType}}
+            @onChange={{fn this.onChangeAccessType "wif"}}
+            @disabled={{this.disableAccessType}}
+          />
+          <label for="access-type-wif">Workload Identity Federation</label>
+        </div>
+      </fieldset>
```

Comment:
Since this is reused in AWS curious why we didn't make a shared component?

Reply:
GCP, AWS and Azure will all have some small overlap because we're adding this "Access Type" option to enterprise views. However, they are some differences, specifically the group values that relate back to these engine's models. 

Additionally, there's a chance Azure and GCP will move to their own engines depending on next steps during the full engine build-outs. I decided to keep all of their logic separate in case future changes. Keeping each configuration create/edit to their own component file with no shared nested components I think makes it easier to test, and/or separate out depending on future steps.